dev/source/docs/mavlink-requesting-data.rst

```diff
@@ -146,6 +146,27 @@ Send a `COMMAND_LONG <https://mavlink.io/en/messages/common.html#COMMAND_LONG>`_
 - param2: depends on message requested; see that message's definition for details.
 - param3 to param7: 0 (not used)
 
+Specifying Message Rates in a File
+----------------------------------
+
+At boot ArduPilot will populate the initial message intervals from files found in either ROMFS or in the filesystem (e.g. SD card).
```

Comment:
at SD root? at APM dir root?

this is really cool....did not know it was in the code....is this master only or in stable? I assume it can be overwritten by the GCS....its a common problem on Dragonlink...I think we dismissed the option of stopping GCS from overwriting, hoping systems they would include it in DL docs, but still is a re-occurring problem for that system

also, if we could create an example for Dragonlink and include it, it would help a lot of people using that system

Reply:
@Hwurzburg this does not stop the GCSs from reconfiguring the telemetry rates using `SET_STREAM_RATE` or whatever.  It just permits absolute customisation of the stream rates for streaming messages, so you can prune out everything that doesn't matter to you.

GCSs should have options to turn off that auto-configuration of stream rates, as @khancyr mentions.  One extreme fix if a GCS doesn't behave is to turn on mavlink signing - can't reset streamrates if you don't know the key!

Reply:
Also, I haven't specified "APM directory" because that's not correct in the Linux HAL.  If you feel like adding more words, I'm happy to think about them :-)

Reply:
what dir in Linux systems?

Reply:
> what dir in Linux systems?

Varies based on board type, but typically the directory in which the binary is started.

Reply:
still needs specific instructions where to place the files for ROMFS or SD...by system if needed...otherwise will just cause support questions...

Reply:
I've added instructions on SD and ROMFS.